tests/bud/base-with-arg/Containerfile2

```diff
@@ -0,0 +1,10 @@
+FROM alpine as build
+
+FROM build AS platform-first
+ENV BUILT_FOR=first
+
+FROM build AS platform-second
+ENV BUILT_FOR=second
+
+FROM platform-${CUSTOM_TARGET} AS final
```

Comment:
I think this happens because we don't pass `Args` from the list of `AllowedArgs` while doing conversion in imagebuilder. But this happens for all the directives where imagebuilder is doing this conversion. Eg `ADD` https://github.com/openshift/imagebuilder/blob/master/dispatchers.go#L145

So for example in case of `ADD` directive we see similar behavior 

```Dockerfile
FROM alpine
ADD --chown=${CUSTOM_USER} buildah .
RUN echo hello
```
and building with `buildah build --build-arg CUSTOM_USER=root -t test .`

I was not aware of this behavior until now but looks like this was always like this in imagebuilder. I think this can be fixed at imagebuilder end for all the directives.

Reply:
When I try this without supplying a CUSTOM_USER value on the command line, I get _no_ error.

Reply:
I think that happens because CUSTOM_USER expands to nothing so empty string there but if I try `buildah build --build-arg CUSTOM_USER=hello -t test .` the build fails with `error building at STEP "ADD --chown=${CUSTOM_USER} buildah .": error looking up UID/GID for "hello": error determining run uid: user: unknown user error looking up user "hello"` cause expansion still happens however it was not supposed to happen cause containerfile never declared any `CUSTOM_USER` using `ARG CUSTOM_USER`. @nalind Could you verify the same ?

Reply:
Yes, confirmed.

Reply:
Created separate issue for this https://github.com/openshift/imagebuilder/issues/231 on imagebuilder so we can fix this for all the instruction where this behavior persists.

Reply:
@nalind Following PR at imagebuilder should handle this issue for all instructions https://github.com/openshift/imagebuilder/pull/232

Reply:
Imagebuilder PR was merged and added a test here as well to verify the same.